src/uagents/relay.py

```diff
@@ -0,0 +1,94 @@
+import asyncio
+import json
+import logging
+
+import aiohttp
+
+from uagents.config import RELAY_POLL_INTERVAL_SECONDS
+from uagents.dispatch import dispatcher
+from uagents.envelope import Envelope
+
+
+class RelayClient:
+    def __init__(self, agent, server_url: str):
+        self._server_url = server_url
+        self._agent = agent
+        self._access_token: str = None
+        self._poll_interval = RELAY_POLL_INTERVAL_SECONDS
+
+    async def run(self):
+        while True:
+            if self._access_token is None:
+                await self._get_access_token()
+            await self._poll_server()
+            await asyncio.sleep(self._poll_interval)
+
+    async def _poll_server(self):
+        async with aiohttp.ClientSession() as session:
+            mailbox_url = f"{self._server_url}/v1/mailbox"
+            async with session.get(
+                mailbox_url,
+                headers={"Authorization": f"token {self._access_token}"},
+            ) as resp:
+                success = resp.status == 200
+                if success:
+                    items = (await resp.json())["items"]
+                    for item in items:
+                        env = Envelope.parse_obj(item["envelope"])
+                        if env.verify():
+                            await dispatcher.dispatch(
+                                env.sender,
+                                env.target,
+                                env.protocol,
+                                env.decode_payload(),
+                            )
+                else:
+                    logging.exception(
+                        f"Failed to retrieve messages from inbox: {(await resp.text())}"
+                    )
+
+            if success and len(items) > 0:
+                async with session.delete(
+                    mailbox_url,
+                    headers={"Authorization": f"token {self._access_token}"},
+                ) as resp:
```

Comment:
Deleting everything in the mailbox is a little dangerous because a subsequent message might have come in. I suggest that you iterate over the message ids and delete  them individually